contracts/test/invariant/InvariantTestERC4626PCVDeposit.t.sol

```diff
@@ -0,0 +1,183 @@
+contract InvariantTestERC4626PCVDeposit is DSTest, DSInvariantTest {
+    using SafeCast for *;
+
+    CoreV2 public core;
+    MockERC20 public token;
+    MockERC4626Vault public tokenizedVault;
+    SystemEntry public entry;
+    PCVGuardian public pcvGuardian;
+    MockPCVOracle public pcvOracle;
+    ERC4626PCVDepositTest public erc4626PCVDepositTest;
+    ERC4626PCVDeposit public tokenizedVaultPCVDeposit;
+
+    Vm private vm = Vm(HEVM_ADDRESS);
+
+    function setUp() public {
+        core = getCoreV2();
+        token = new MockERC20();
+        pcvOracle = new MockPCVOracle();
+        tokenizedVault = new MockERC4626Vault(MockERC20(address(token)));
+        tokenizedVaultPCVDeposit = new ERC4626PCVDeposit(
+            address(core),
+            address(token),
+            address(tokenizedVault),
+            true
+        );
+
+        address[] memory toWhitelist = new address[](1);
+        toWhitelist[0] = address(tokenizedVaultPCVDeposit);
+
+        pcvGuardian = new PCVGuardian(
+            address(core),
+            address(this),
+            toWhitelist
+        );
+
+        entry = new SystemEntry(address(core));
+        erc4626PCVDepositTest = new ERC4626PCVDepositTest(
+            tokenizedVaultPCVDeposit,
+            token,
+            tokenizedVault,
+            entry,
+            pcvGuardian
+        );
+
+        vm.startPrank(addresses.governorAddress);
+
+        tokenizedVaultPCVDeposit.setPCVOracle(address(pcvOracle));
+
+        core.grantPCVGuard(address(erc4626PCVDepositTest));
+        core.grantPCVController(address(pcvGuardian));
+
+        core.grantLocker(address(pcvGuardian));
+        core.grantLocker(address(tokenizedVaultPCVDeposit));
+
+        vm.stopPrank();
+
+        addTargetContract(address(erc4626PCVDepositTest));
+    }
+
+    function invariantLastRecordedBalance() public {
+        assertEq(
+            tokenizedVaultPCVDeposit.lastRecordedBalance(),
+            erc4626PCVDepositTest.totalDeposited()
+        );
+        assertEq(
+            tokenizedVaultPCVDeposit.balance(),
+            erc4626PCVDepositTest.totalDeposited()
+        );
+    }
+
+    function invariantPcvOracle() public {
+        assertEq(
+            tokenizedVaultPCVDeposit.lastRecordedBalance(),
+            pcvOracle.pcvAmount().toUint256()
+        );
+        assertEq(
+            tokenizedVaultPCVDeposit.lastRecordedBalance(),
+            tokenizedVaultPCVDeposit.balance()
+        );
+        assertEq(
+            tokenizedVaultPCVDeposit.balance(),
+            erc4626PCVDepositTest.totalDeposited()
+        );
+    }
+
+    function invariantBalanceOf() public {
+        assertEq(
+            tokenizedVaultPCVDeposit.balance(),
+            tokenizedVault.convertToAssets(
+                tokenizedVault.balanceOf(address(tokenizedVaultPCVDeposit))
+            )
+        );
```

Comment:
invariant tests are looking good! would be great to add one additional assertion here that `convertToAssets` matches pcvOracle `pcvAmount`